orchestrator/relayer/src/main_loop.rs

```diff
@@ -23,42 +28,149 @@ pub const ETH_SUBMIT_WAIT_TIME: Duration = Duration::from_secs(600);
+    // Update the tracker with the now-known desired number of samples
+    update_gas_history_samples(config.altruistic_gas_price_samples as usize);
+    debug!("Starting all relayer loops");
+
+    let a = relayer_main_loop(
+        ethereum_key,
+        web3.clone(),
+        grpc_client.clone(),
+        gravity_contract_address,
+        gravity_id,
+        config.clone(),
+    );
+    let b = ibc_auto_forward_loop(
+        cosmos_key,
+        &contact,
+        grpc_client.clone(),
+        fee.clone(),
+        config.clone(),
+    );
+    let c = gas_tracker_loop(&web3, config.clone());
+
+    if let (Some(key), Some(cosmos_fee)) = (cosmos_key, fee) {
+        let d = batch_request_loop(
+            contact.clone(),
+            &web3,
+            grpc_client.clone(),
+            config.clone(),
+            ethereum_key,
+            key,
+            cosmos_fee.clone(),
+        );
+
+        join4(a, b, c, d).await;
+    } else {
+        join3(a, b, c).await;
+    }
+}
 
 /// This function contains the relayer primary loop, it is broken out of the main loop so that
 /// it can be called in the test runner for easier orchestration of multi-node tests
 #[allow(clippy::too_many_arguments)]
 pub async fn relayer_main_loop(
     ethereum_key: EthPrivateKey,
-    cosmos_key: Option<CosmosPrivateKey>,
-    cosmos_fee: Option<Coin>,
     web3: Web3,
-    contact: Contact,
     grpc_client: GravityQueryClient<Channel>,
     gravity_contract_address: EthAddress,
     gravity_id: String,
     relayer_config: RelayerConfig,
 ) {
+    {
+        // Initialize RELAYER_START via the very first deref
+        let start = *RELAYER_START;
+        debug!("Relayer Started at {:?}", start);
+    }
     let mut grpc_client = grpc_client;
 
-    let mut gas_estimator = GasTracker::new(ALTRUISTIC_SAMPLES);
+    // Separately the batch_request_loop should be running immediately, we delay this main loop in order to
+    // prevent newly requested batches from being temporarily unrelayable. Batch requests take 20-25 seconds to process,
+    // meaning we could miss new ones, taking 10+ minutes to relay it
+    delay_for(Duration::from_secs(
+        relayer_config.batch_request_relay_offset,
+    ))
+    .await;
+
     loop {
         let loop_start = Instant::now();
 
-        // update the gas estimator and determine if we should relay altruistically
-        let current_gas_price = gas_estimator.update(&web3).await;
-        let ideal_gas = gas_estimator.get_acceptable_gas_price(ALTRUISTIC_GAS_PERCENTAGE);
-        let should_relay_altruistic =
-            if let (Some(current_price), Some(good_price)) = (current_gas_price, ideal_gas) {
-                current_price <= good_price
-            } else {
-                false
-            };
+        // use the gas estimator to determine if we should relay altruistically
+        // TODO: Check to see when the gas price was sampled? Only a significant issue if the gas tracker loop speed is large
```

Comment:
maybe add an assert that the tracker loop speed is less than say one minute, should keep anyone from setting it to an hour and getting ridiculous results.